### PR TITLE
Pipewire: add factory Create method

### PR DESCRIPTION
**Pull request by lrusak**
Currently pipewire is tied to `CAESinkPipewire` in AudioEngine. This PR breaks that tie and allows a new pipewire instance to be created through the new `CPipewire::Create()` factory method. This will be useful in the future if we have multiple instances of pipewire running.

This also moves the try/catch to be more specific about where we throw (we really only expect it to happen when connecting to the pipewire server.

The only gotcha that I see is `pw_deinit` as the docs state:
> Before 0.3.49 this function can only be called once after which the pipewire library can not be used again. This is usually called by test programs to check for memory leaks.

> Since 0.3.49 this function must be paired with an equal amount of [pw_init()](https://docs.pipewire.org/group__pw__pipewire.html#ga06c879b2d800579f4724109054368d99) calls to deinitialize the PipeWire library. The PipeWire library can be used again after being deinitialized with a new [pw_init()](https://docs.pipewire.org/group__pw__pipewire.html#ga06c879b2d800579f4724109054368d99) call.

ref: https://docs.pipewire.org/group__pw__pipewire.html#gafa6045cd7391b467af4575c6752d6c4e

Currently we only have once pipewire instance so this shouldn't matter but if we add more we should bump the minimum pipewire version to `0.3.49`

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/cores/AudioEngine/Sinks/pipewire/Pipewire.cpp [hunk 1]
```diff
@@ -68,10 +78,24 @@ bool CPipewire::Start()
...
   }
 
   return true;
 }
+
+std::unique_ptr<CPipewire> CPipewire::Create()
+{
+  struct PipewireMaker : public CPipewire
+  {
+    using CPipewire::CPipewire;
```
Comment (garbear) [inline xbmc/cores/AudioEngine/Sinks/pipewire/Pipewire.cpp:92]: This must allow for adding functionality to PipewireMaker in the future?

Comment (lrusak) [reply xbmc/cores/AudioEngine/Sinks/pipewire/Pipewire.cpp:92]: It's purely for allowing the use of a private constructor with `std::make_unique`

**Comment by lrusak** [comment]
jenkins build this please

**Pull request merged**